R/utilities.R

```diff
@@ -0,0 +1,540 @@
+                pull_hifld_field(HIFLD.ID, "COUNTY", hifld_data)), 
+            County.FIPS = coalesce(
+                County.FIPS, 
+                pull_hifld_field(HIFLD.ID, "COUNTYFIPS", hifld_data)), 
+            Capacity = coalesce(
+                Capacity, 
+                pull_hifld_field(HIFLD.ID, "CAPACITY", hifld_data)), 
+            Source.Capacity = coalesce(
+                Source.Capacity, 
+                pull_hifld_field(HIFLD.ID, "capacity_source", hifld_data)), 
+            Website = coalesce(
+                Website, 
+                pull_hifld_field(HIFLD.ID, "WEBSITE", hifld_data)), 
+            Latitude = coalesce(
+                Latitude, 
+                pull_hifld_field(HIFLD.ID, "lat", hifld_data)), 
+            Longitude = coalesce(
+                Longitude, 
+                pull_hifld_field(HIFLD.ID, "lon", hifld_data))
+        ) %>% 
+        ungroup() %>%
+        
+        # Geocode addresses
+        tidygeocoder::geocode(
+            method = "geocodio", 
+            street = Address,
+            city = City,
+            county = County,
+            state = State,
+            postalcode = Zipcode,
+            lat = "lat_geo_",
+            long = "long_geo_"
+        ) %>%
+        mutate(
+            Latitude = coalesce(Latitude, lat_geo_),
+            Longitude = coalesce(Longitude, long_geo_)
+        ) %>%
+        select(-lat_geo_, -long_geo_) %>%
+        
+        # Assign Facility.ID
+        mutate(Facility.ID = generate_new_fac_id(
+            old_fac_info = old_fac_info, 
+            old_fac_spellings = old_fac_spellings) + row_number() - 1)
+}
+
+populate_new_spellings <- function(
+    new_fac_spellings = NULL, old_fac_info = NULL) {
+    #' Populate new facility spellings  
+    #' 
+    #' Assigns the facility ID to each spelling based on the facility info data.  
+    #' 
+    #' @param new_fac_spellings data frame, new facility spelling updates
+    #' @param old_fac_info data frame, existing facility info  
+    
+    if (is.null(new_fac_spellings)) {
+        new_fac_spellings <- read_new_fac_spellings()
+    }
+    if (is.null(old_fac_info)) {
+        old_fac_info <- read_fac_info()
+    }
+    
+    # Merge federal facilities on cleaned name 
+    federal <- new_fac_spellings %>% 
+        filter(Is.Federal == 1) %>% 
+        mutate(name_clean_ = clean_fac_col_txt(xwalk_name_clean, to_upper = TRUE)) %>% 
+        select(-State) %>% 
+        left_join(old_fac_info %>% 
+                      filter(Jurisdiction == "federal") %>%  
+                      mutate(name_clean_ = clean_fac_col_txt(Name, to_upper = TRUE)),  
+                  by = "name_clean_") %>% 
+        select(Facility.ID, xwalk_name_raw, xwalk_name_clean, State, Is.Federal, Source)
+    
+    # Merge non-federal facilities on cleaned name AND state 
+    non_federal <- new_fac_spellings %>% 
+        filter(!Is.Federal %in% c(1)) %>% # Is.Federal = NA are treated as non-federal 
```

Comment:
is this the same as `Is.Federal == 0` ? Curious why you chose to write it like this

Reply:
I think its because we want both NA and 0 which maybe writing `Is.Federal %in% c(0, NA)` is more clear. When do we have an instance where we dont know federal status tho? Shouldnt we always know that?

Reply:
Yup! It's for the NAs, but indeed multiple double-negatives in one conditional is Very Confusing. I'll update it now! And I think we'll always know, but maybe not for historical data? Also, none of the facilities currently in the [Google Sheet](https://docs.google.com/spreadsheets/d/1tAhD-brnuqw0s55QXM-xYLPsyD-rNrqHbAVIbxSOMwI/edit#gid=0) have an `Is.Federal` flag, so I figured I'd treat anything not a 1 as a 0 for now.